smdebug/core/hook.py

```diff
@@ -383,6 +393,19 @@ def _close_writers(self) -> None:
     def _initialize_writers(self) -> None:
         if self.dry_run:
             return
+        if self.first_process is False:
+            return
+        elif self.first_process is None:
+            if self._get_num_workers() == 1:
```

Comment:
check for TF _get_num_workers , if unsupported it will error out?

Reply:
It doesn't error out with the code changes this PR brings in.